dart/test/mocks/mock_hub.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:sentry/sentry.dart';
+
+class MockHub implements Hub {
+  List<CaptureEventCall> captureEventCalls = [];
+  List<CaptureExceptionCall> captureExceptionCalls = [];
+  List<CaptureMessageCall> captureMessageCalls = [];
+  List<AddBreadcrumbCall> addBreadcrumbCalls = [];
+  List<SentryClient?> bindClientCalls = [];
+  int closeCalls = 0;
+
+  @override
+  void addBreadcrumb(Breadcrumb crumb, {dynamic hint}) {
+    addBreadcrumbCalls.add(AddBreadcrumbCall(crumb, hint));
+  }
+
+  @override
+  void bindClient(SentryClient client) {
+    bindClientCalls.add(client);
+  }
+
+  @override
+  Future<SentryId> captureEvent(
+    SentryEvent event, {
+    dynamic stackTrace,
+    dynamic hint,
+  }) async {
+    captureEventCalls.add(CaptureEventCall(event, stackTrace, hint));
+    return event.eventId;
+  }
+
+  @override
+  Future<SentryId> captureException(
+    dynamic throwable, {
+    dynamic stackTrace,
+    dynamic hint,
+  }) async {
+    captureExceptionCalls
+        .add(CaptureExceptionCall(throwable, stackTrace, hint));
+    return SentryId.newId();
+  }
+
+  @override
+  Future<SentryId> captureMessage(
+    String? message, {
+    SentryLevel level = SentryLevel.info,
+    String? template,
+    List? params,
+    dynamic hint,
+  }) async {
+    captureMessageCalls
+        .add(CaptureMessageCall(message, level, template, params, hint));
+    return SentryId.newId();
+  }
+
+  @override
+  Hub clone() {
+    // TODO: implement clone
+    throw UnimplementedError();
+  }
+
+  @override
+  void close() {
+    closeCalls = closeCalls + 1;
+  }
+
+  @override
+  void configureScope(callback) {
+    // TODO: implement configureScope
+  }
+
+  @override
+  // TODO: implement isEnabled
+  bool get isEnabled => throw UnimplementedError();
+
+  @override
+  // TODO: implement lastEventId
+  SentryId get lastEventId => throw UnimplementedError();
+}
```

Comment:
we had tests for this before, not anymore, I think we should work on them or the chance to add a regression is higher

Reply:
I am not quite sure which test you refer to. All of those unimplemented things were also previously not verified with the mocks through mockito. Or am I overlooking something?
The only test which does something similar is https://github.com/getsentry/sentry-dart/blob/e8fbf20c673e3e9f85b697c3151d00c2b9a24a1c/dart/test/hub_test.dart#L90 but it is for the SentryClient

Reply:
`hub_test` has tests with `expect(hub.isEnabled...`

`hub_test` has tests with `hub.configureScope...`

and the one you've mentioned, `should save the lastEventId`

Reply:
The one I mentioned is for the `SentryClient` tho. 

The following tests 

https://github.com/getsentry/sentry-dart/blob/e8fbf20c673e3e9f85b697c3151d00c2b9a24a1c/dart/test/hub_test.dart#L177 

https://github.com/getsentry/sentry-dart/blob/e8fbf20c673e3e9f85b697c3151d00c2b9a24a1c/dart/test/hub_test.dart#L34

https://github.com/getsentry/sentry-dart/blob/e8fbf20c673e3e9f85b697c3151d00c2b9a24a1c/dart/test/hub_test.dart#L113 

https://github.com/getsentry/sentry-dart/blob/e8fbf20c673e3e9f85b697c3151d00c2b9a24a1c/dart/test/hub_test.dart#L141 

aren't using the mock.

Reply:
that's true :)